Framework/Algorithms/src/CopySample.cpp

```diff
@@ -150,6 +158,52 @@ void CopySample::copyParameters(Sample &from, Sample &to, bool nameFlag, bool ma
       rhsMaterial = to.getMaterial();
     }
     auto rhsObject = std::shared_ptr<IObject>(from.getShape().cloneWithMaterial(rhsMaterial));
+
+    if (auto csgObj = std::dynamic_pointer_cast<Geometry::CSGObject>(rhsObject)) {
+      // Rotate CSGObject by goniometer by editing XML, if possible for that shape
+
+      std::string xml = rhsObject->getShapeXML();
```

Comment:
If you did csgObj->getShapeXML() here would that remove the need to define the getShapeXML method on all of the IObject classes?